java/semantickernel-api/src/main/java/com/microsoft/semantickernel/semanticfunctions/KernelPromptTemplateFactory.java

```diff
@@ -1,9 +1,21 @@
 package com.microsoft.semantickernel.semanticfunctions;
 
-public class KernelPromptTemplateFactory implements PromptTemplateFactory{
+import com.microsoft.semantickernel.templateengine.handlebars.HandlebarsPromptTemplate;
+import com.microsoft.semantickernel.templateengine.semantickernel.DefaultPromptTemplate;
+import java.util.Locale;
+
+public class KernelPromptTemplateFactory implements PromptTemplateFactory {
 
     @Override
     public PromptTemplate tryCreate(PromptTemplateConfig templateConfig) {
-        return null;
+        switch (templateConfig.getTemplateFormat().toLowerCase(Locale.ROOT)) {
```

Comment:
Move this to PromptTemplateConfig as a static method and get rid of the factory. 

In dotnet, they have "boolean tryCreate" which doesn't work in java. Maybe this should throw an "UnknownTemplateException"

Reply:
Unfortunately removing this factory would  have quite a few impacts that would move us away from .net